### PR TITLE
Enhance UX for managing credentials

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

This PR refactors multiple context menu items:
* Update Credentials
* Log In to Authentication Service
* Log Out of Authentication Service

into a single menu item "Manage Credentials" which shows a quickpick for users to choose an authentication method.

image

## Release Notes

Milestone: 2.12.0

Changelog:
> Replaced multiple options for updating credentials with a single "Manage Credentials" option that prompts for preferred authentication type. [#2263](https://github.com/zowe/vscode-extension-for-zowe/issues/2263)

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [x] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

In addition to improving UX, this should make it easier to support certificates (#2373).

### CODE REVIEW

**Review by traeok — DISMISSED** [review]
LGTM, thanks @t1m0thyj 😁

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 ### New features and enhancements
 
 - Allow deleting migrated datasets [#2447](https://github.com/zowe/vscode-extension-for-zowe/issues/2447)
+- Replaced multiple options for updating credentials with a single "Manage Credentials" option that prompts for preferred authentication type. [#2263](https://github.com/zowe/vscode-extension-for-zowe/issues/2263)
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/CHANGELOG.md:16]: should move CHANGELOG item under TBD

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
this is looking great @t1m0thyj!!
I was wondering if the logout option should be a radio button too? maybe not since the others are showing what is in use. Should we use a separator line for logout then similar to add/create profile?
Screen Shot 2023-09-15 at 4 44 22 PM



I also notice that the quickpick didn't close after choosing the logout option, I have to escape it.
Screen Shot 2023-09-15 at 4 38 30 PM

**Review by JillieBeanSim — APPROVED** [review]
Thanks for pushing the latest updates, everything works great! Thanks @t1m0thyj for this enhancement

**Review by rudyflores — CHANGES_REQUESTED** [review]
When performing a token auth login, seems to be working well, I do have a concern in terms of the UI/UX for logging out in this change, should we show instead just a "logout" button or show the option for logout as a radio button too? i'm curious why it has a different UI for the logout.

Another thing I wanted to mention is that when clicking "logout" the pop up window with the option doesn't close but remains open, but the bottom right show that logout was done successfully, my feedback on this is I felt like maybe there is no feedback when clicking on the button that will tell the user that they logged out unless they look at the bottom right, I think maybe closing the pop up selection menu will help with this!

Screenshot 2023-09-18 at 11 53 03 AM

**Comment by rudyflores** [comment]
One more thing @t1m0thyj 

if you perform the following on a z/OSMF or extender profile:
1. Log in with token auth
2. click manage credentials and choose basic auth
3. type in credentials for basic auth

This causes now the profile to be using both basic and token auth which now doesn't allow the user to logout unless the secure array properties with `user` and `password` are removed, I think we may not want to mix credential management types here but may be something we can discuss as well.

**Comment by t1m0thyj** [comment]
@rudyflores Thanks for the feedback! 😍 

> When performing a token auth login, seems to be working well, I do have a concern in terms of the UI/UX for logging out in this change, should we show instead just a "logout" button or show the option for logout as a radio button too? i'm curious why it has a different UI for the logout.

The radio buttons are for authentication options (User and Password, Authentication Token, and in the future Certificate File). Since logout is an action separate from these, I don't think it should be a radio button. I agree a "logout" button would be nice but unfortunately the UI options provided by VS Code for quickpicks are limited so we can't show a button with text on its own row. I've mocked up some alternative ideas:
1. Logout icon shown on the right of the quickpick - this is the only style of button they support. The button can show a tooltip when hovered, but I'm not sure if its obvious enough for users to easily find the "logout" option.
    image
2. Adding an icon to the existing "logout" option - this is a small change but I think it looks better.
    image

> Another thing I wanted to mention is that when clicking "logout" the pop up window with the option doesn't close but remains open, but the bottom right show that logout was done successfully, my feedback on this is I felt like maybe there is no feedback when clicking on the button that will tell the user that they logged out unless they look at the bottom right, I think maybe closing the pop up selection menu will help with this!

Agree that the popup should close after logout, did you test with the latest commit? There was an issue with the popup not being closed that Billie pointed out which should be fixed in https://github.com/zowe/vscode-extension-for-zowe/pull/2445/commits/5a6fc3abdedf6d3503469b40820fcd00db82fefc

**Comment by JillieBeanSim** [comment]
Hey @t1m0thyj due to concerns with the authentication methods and pre-existing issues with offering basic auth info gathering while token auth in place and other issues I have started work on a [POC](https://github.com/zowe/vscode-extension-for-zowe/pull/2472) for profile management based off of these ideas, keep in mind this isn't complete yet.  We can continue discussions on how we would like to proceed

**Pull request closed without merging**